E2E: selectively apply extended timeout in specific cases.

worldomonation opened the pull request:
#### Proposed Changes

This PR deals with the minor fallout of https://github.com/Automattic/wp-calypso/pull/70392.

Key changes:
- extend timeouts from the default 10s at key areas that require longer timeouts.
- remove manual override of timeouts (`10 * 1000`) because the 10s timeout is now default.

#### Testing Instructions

Ensure the following build configurations are passing:
  - [x] Calypso E2E (desktop)
  - [x] Calypso E2E (mobile)

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Caused by https://github.com/Automattic/wp-calypso/pull/70392.
Supersedes https://github.com/Automattic/wp-calypso/pull/70561.

worldomonation committed:
Add extended timeout to select `waitForLoadState` calls

worldomonation committed:
Remove 10,000ms tiemout overrides in specs and POMs

worldomonation committed:
packages/calypso-e2e/src/lib/components/sidebar-component.ts
- add extended wait for `SidebarInitialization`.

packages/calypso-e2e/src/lib/pages/editor-page.ts
- add extended wait for `visit` method.

worldomonation committed:
packages/calypso-e2e/src/lib/pages/editor-page.ts
- extend the `visit` timeout to 30 seconds

worldomonation committed:
packages/calypso-e2e/src/lib/components/editor-gutenberg-component.ts
- extend timeout for `enterTitle` method

worldomonation committed:
Additional 20 * 1000 timeout on select `networkidle` waits.

worldomonation commented on test/e2e/specs/i18n/i18n__editor.ts:
```diff
@@ -296,19 +296,14 @@ describe( 'I18N: Editor', function () {
-				await frame
-					.locator( etkTranslations.welcomeGuide.closeButtonSelector )
-					.click( { timeout } );
+				await frame.locator( etkTranslations.welcomeGuide.openGuideSelector ).click();
+				await frame.waitForSelector( etkTranslations.welcomeGuide.welcomeTitleSelector );
+				await frame.locator( etkTranslations.welcomeGuide.closeButtonSelector ).click();
```
Removal of the timeout override.

worldomonation commented on test/e2e/specs/i18n/i18n__editor.ts:
```diff
@@ -333,9 +326,7 @@ describe( 'I18N: Editor', function () {
-							frame.waitForSelector( `${ block.blockEditorSelector } ${ content }`, {
-								timeout: blockTimeout,
-							} )
+							frame.waitForSelector( `${ block.blockEditorSelector } ${ content }` )
```
Removal of the timeout override.

worldomonation commented on test/e2e/specs/i18n/i18n__editor.ts:
```diff
@@ -346,8 +337,7 @@ describe( 'I18N: Editor', function () {
 					await frame.waitForSelector(
-						`:is( ${ block.blockEditorSelector }.is-selected, ${ block.blockEditorSelector }.has-child-selected)`,
-						{ timeout: blockTimeout }
+						`:is( ${ block.blockEditorSelector }.is-selected, ${ block.blockEditorSelector }.has-child-selected)`
```
Removal of the timeout override.

worldomonation committed:
test/e2e/specs/fse/fse__temp-smoke-test.ts
- extend the `waitForLoadState` in the last step to 30 seconds

worldomonation commented on packages/calypso-e2e/src/lib/components/sidebar-component.ts:
```diff
@@ -35,9 +35,12 @@ export class SidebarComponent {
-		await this.page.waitForLoadState( 'load' );
 		const sidebarLocator = this.page.locator( selectors.sidebar );
-		await sidebarLocator.waitFor();
+
+		await Promise.all( [
+			this.page.waitForLoadState( 'load', { timeout: 20 * 1000 } ),
+			sidebarLocator.waitFor( { timeout: 20 * 1000 } ),
+		] );
```
@karenroldan This should take care of the SidebarComponent timeout you mentioned in https://github.com/Automattic/wp-calypso/pull/70561.

karenroldan reviewed (commented):
**praise:** thank you so much for working on this and improving the execution time! 🥳 

One more thing...
**suggestion:** I have rerun the build for i8n tests but it still fails due to 10s timeout.  The element from the sidebar is detached from the DOM. It could be because of the upsell banner that shows up in the sidebar a bit later.

Could we also see if extending the timeout for the `MeSidebarComponent` or waiting until `networkidle` resolves it? 

**issue:** Also, there is another failure: `Build failure on metric change` -- I am not sure if the issue is related to the timeout also.

**todo:** please resolve the conflicts too. 👍🏻

worldomonation commented:
> issue: Also, there is another failure: Build failure on metric change -- I am not sure if the issue is related to the timeout also.

This is because we have a rule in TeamCity that fails a build if less than a certain number of tests pass compared to the last successful run. In this case, this was triggered by the timeout on the component you identified earlier, so once that is fixed this error will also go away.

> todo: please resolve the conflicts too. 👍🏻

Ah yeah, fun times.

> suggestion: I have rerun the build for i8n tests but it still fails due to 10s timeout. The element from the sidebar is detached from the DOM. It could be because of the upsell banner that shows up in the sidebar a bit later.

I'll double-check the I18N build configuration, and while I'm at it will also check the other suites we have that don't run regularly like Gutenberg builds.

worldomonation committed:
test/e2e/specs/i18n/i18n__editor.ts
- replace the use of ChangeUIFlow with RestAPIClient.

test/e2e/specs/martech/tos-screenshots__checkout.ts
- pre-emptively add extended timeouts to networkidle waits.

worldomonation commented on test/e2e/specs/i18n/i18n__editor.ts:
```diff
@@ -265,50 +266,33 @@ describe( 'I18N: Editor', function () {
-					page.goto( DataHelper.getCalypsoURL( '/' ) ),
-				] );
-			} );
+		beforeAll( async function () {
+			await restAPIClient.setMySettings( { language: locale } );
+			await page.reload( { waitUntil: 'networkidle', timeout: 20 * 1000 } );
+		} );
```
@karenroldan The solution to the issue encountered in `ChangeUILanguageFlow` is to use the API to set the language.

We use the API throughout the framework already and despite it not being entirely representative of user interaction, I believe it's acceptable here because we're not testing the language-changing flow but rather that strings on the page are updated as expected.

Let me know if you agree with this solution!

karenroldan replied:
> I believe it's acceptable here because we're not testing the language-changing flow but rather that strings on the page are updated as expected.

I agree with this. It's an efficient way to ensure that translations are working properly. Thanks for working on this!

karenroldan replied:
> I believe it's acceptable here because we're not testing the language-changing flow but rather that strings on the page are updated as expected.

I agree with this. It's an efficient way to ensure that translations are working properly. Thanks for working on this!

worldomonation replied:
At some point, I will extract this logic into a helper function as this same snippet is repeated elsewhere at least once.

worldomonation committed:
packages/calypso-e2e/src/lib/flows/start-site-flow.ts
- extend timeout to 20s.
